sdk/core/azure-core/src/main/java/com/azure/core/http/rest/ErrorOptions.java

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.rest;
+
+/**
+ * Determines how errors are handled by requests using {@link RequestOptions}.
+ */
+public enum ErrorOptions {
+    /**
+     * Throw exceptions when an HTTP response with a status code indicating an error (400 or above) is received.
+     */
+    THROW,
```

Comment:
would we ever need to handle 500 or above differently than 400-499?

Reply:
That's a great question, the short answer would be maybe. Though, this aligns with default handling that already occurs elsewhere within the SDKs.